deeppavlov/models/bert/bert_as_summarizer.py

```diff
@@ -0,0 +1,213 @@
+                              is_training=self.is_train_ph,
+                              input_ids=self.input_ids_ph,
+                              input_mask=self.input_masks_ph,
+                              token_type_ids=self.token_types_ph,
+                              use_one_hot_embeddings=False,
+                              )
+        # next sentence prediction head
+        with tf.variable_scope("cls/seq_relationship"):
+            output_weights = tf.get_variable(
+                "output_weights",
+                shape=[2, self.bert_config.hidden_size],
+                initializer=create_initializer(self.bert_config.initializer_range))
+            output_bias = tf.get_variable(
+                "output_bias", shape=[2], initializer=tf.zeros_initializer())
+
+        nsp_logits = tf.matmul(self.bert.get_pooled_output(), output_weights, transpose_b=True)
+        nsp_logits = tf.nn.bias_add(nsp_logits, output_bias)
+        self.nsp_probs = tf.nn.softmax(nsp_logits, axis=-1)
+
+    def _init_placeholders(self):
+        self.input_ids_ph = tf.placeholder(shape=(None, None), dtype=tf.int32, name='ids_ph')
+        self.input_masks_ph = tf.placeholder(shape=(None, None), dtype=tf.int32, name='masks_ph')
+        self.token_types_ph = tf.placeholder(shape=(None, None), dtype=tf.int32, name='token_types_ph')
+
+        self.is_train_ph = tf.placeholder_with_default(False, shape=[], name='is_train_ph')
+
+    def _build_feed_dict(self, input_ids, input_masks, token_types):
+        feed_dict = {
+            self.input_ids_ph: input_ids,
+            self.input_masks_ph: input_masks,
+            self.token_types_ph: token_types,
+        }
+        return feed_dict
+
+    def _get_nsp_predictions(self, sentences: List[str], candidates: List[str]):
+        """Compute NextSentence probability for every (sentence_i, candidate_i) pair.
+
+        [CLS] sentence_i [SEP] candidate_i [SEP]
+
+        Args:
+            sentences (List[str]): list of sentences
+            candidates (List[str]): list of candidates to be the next sentence
+
+        Returns:
+            probabilities that candidate is a next sentence
+        """
+        features = self.bert_preprocessor(texts_a=sentences, texts_b=candidates)
+        input_ids = [f.input_ids for f in features]
+        input_masks = [f.input_mask for f in features]
+        input_type_ids = [f.input_type_ids for f in features]
+        feed_dict = self._build_feed_dict(input_ids, input_masks, input_type_ids)
+        nsp_probs = self.sess.run(self.nsp_probs, feed_dict=feed_dict)
+        return nsp_probs[:, 0]
+
+    def __call__(self, texts: List[str], init_sentences: Optional[List[str]] = None) -> List[List[str]]:
+        """Builds summary for text from `texts`
+
+        Args:
+            texts (List[str]): texts to build summaries for
+            init_sentences (Optional[List[str]], optional): `init_sentence` is used as the first sentence in summary.
```

Comment:
```suggestion
            init_sentences (Optional[List[str]], optional): ``init_sentence`` is used as the first sentence in summary.
```
everywhere for sphinx rendering